include/swift/Sema/ConstraintSystem.h

```diff
@@ -2766,7 +2766,8 @@ class ConstraintSystem {
   void
   filterSolutions(SmallVectorImpl<Solution> &solutions,
                   bool minimize = false) {
-    if (solutions.size() < 2)
+    if (solutions.size() < 2 ||
+        solutions[0].getConstraintSystem().isForCodeCompletion())
```

Comment:
Nit: This is method on constraint system itself so you could simply use `isForCodeCompletion()`.